devito/ir/iet/nodes.py

```diff
@@ -1430,8 +1438,20 @@ def DummyExpr(*args, init=False):
 
 # Nodes required for distributed-memory halo exchange
 
+class HaloMixin:
+
+    def __repr__(self):
+        fstrings = []
+        for f in self.fmapper.keys():
+            loc_indices = OrderedSet(*(self.fmapper[f].loc_indices.values()))
+            loc_indices_str = str(list(loc_indices)) if loc_indices else ""
+            fstrings.append("%s%s" % (f.name, loc_indices_str))
 
-class HaloSpot(Node):
+        functions = ",".join(fstrings)
+        return "<%s(%s)>" % (self.__class__.__name__, functions)
+
+
+class HaloSpot(HaloMixin, Node):
```

Comment:
after re-thinking about this change, I struggle to understand.

* HaloScheme (in `mpi.py`) has this new `__repr__`
* HaloSpot (here) carries a `halo_scheme`
* Hence, couldn't `HaloSpot.__repr__` simply resort to `self.halo_scheme.__repr__()` (or a wrapper of it if you want to add more information)

I think this thing that the same exact representation is used in two different places doesn't make much sense